src/algorithms.jl

```diff
@@ -0,0 +1,234 @@
+abstract type IceFloeTrackerAlgorithm end
```

Comment:
This should be an IceFloeSegementationAlgorithm

Would we need an IceFloeTrackerAlgorithm?
Would we need an IceFloeSegmentationTrackingAlgorithm?